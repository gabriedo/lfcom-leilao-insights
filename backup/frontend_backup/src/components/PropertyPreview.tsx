import React from 'react';

interface PropertyPreviewProps {
  minBid?: string | number;
}

const PropertyPreview: React.FC<PropertyPreviewProps> = ({ minBid }) => {
  const formatCurrency = (value: string | number | undefined) => {
    if (!value) return 'Não informado';
    
    // Converte para número e divide por 100 para corrigir a escala
    const numericValue = Number(value) / 100;
    
    return numericValue.toLocaleString('pt-BR', {
      style: 'currency',
      currency: 'BRL'
    });
  };

  return (
    <span>
      {formatCurrency(minBid)}
    </span>
  );
};

export default PropertyPreview; 