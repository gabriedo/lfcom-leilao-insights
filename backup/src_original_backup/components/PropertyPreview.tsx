import React from 'react';
import { Link } from 'react-router-dom';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { Button } from '@/components/ui/button';
import { Alert, AlertDescription } from '@/components/ui/alert';
import { AlertCircle, Building2, Home, Landmark, RefreshCw } from 'lucide-react';
import { Badge } from '@/components/ui/badge';
import { Separator } from '@/components/ui/separator';
import {
  PropertyHeader,
  PropertyImages,
  PropertyPricing,
  PropertyLocation,
  PropertyExtras
} from './property-preview';

interface PropertyPreviewProps {
  id?: string | null;
  title: string;
  address?: string | null;
  city: string;
  state: string;
  minBid?: string | null;
  evaluatedValue: string;
  propertyType?: string | null;
  auctionType?: string | null;
  auctionDate?: string | null;
  description?: string | null;
  images?: string[] | null;
  documents?: any[] | null;
  auctions?: any[] | null;
  extractionStatus?: 'success' | 'fallback_used' | 'partial' | 'failed';
  onRefresh?: () => void;
}

const statusColors = {
  success: 'default',
  fallback_used: 'secondary',
  partial: 'outline',
  failed: 'destructive'
} as const;

const statusLabels = {
  success: 'Sucesso',
  fallback_used: 'Fallback',
  partial: 'Parcial',
  failed: 'Falha'
} as const;

const getStatusMessage = (status: string | undefined, description?: string | null) => {
  if (status === 'failed') {
    return description || 'Não foi possível extrair os dados deste imóvel. Tente novamente ou verifique a URL.';
  }
  if (status === 'partial') {
    return 'Algumas informações não foram extraídas corretamente. Confira os campos abaixo.';
  }
  return null;
};

const PropertyPreview: React.FC<PropertyPreviewProps> = (props) => {
  console.log('[PropertyPreview] Props recebidas:', {
    title: props.title,
    address: props.address,
    city: props.city,
    state: props.state,
    propertyType: props.propertyType,
    minBid: props.minBid,
    evaluatedValue: props.evaluatedValue,
    extractionStatus: props.extractionStatus,
    documents: props.documents?.length,
    auctions: props.auctions?.length
  });
  
  const {
    id = 'temp-id',
    title = 'Título não disponível',
    address = 'Endereço não disponível',
    city = 'Cidade não informada',
    state = 'Estado não informado',
    minBid = '',
    evaluatedValue = '',
    propertyType = 'Não especificado',
    auctionType = 'Leilão',
    auctionDate = '',
    description = 'Sem descrição disponível',
    images = [],
    documents = [],
    auctions = [],
    extractionStatus = 'success',
    onRefresh
  } = props;

  const statusMessage = getStatusMessage(extractionStatus, description);
  const documentCount = documents?.length ?? 0;
  const bidCount = auctions?.length ?? 0;

  const getPropertyTypeIcon = () => {
    const type = propertyType?.toLowerCase() || '';
    if (type.includes('apartamento') || type.includes('apartment')) {
      return <Building2 className="h-4 w-4" />;
    }
    if (type.includes('casa') || type.includes('house')) {
      return <Home className="h-4 w-4" />;
    }
    if (type.includes('comercial') || type.includes('commercial')) {
      return <Landmark className="h-4 w-4" />;
    }
    return <Home className="h-4 w-4" />;
  };

  const getPropertyTypeLabel = () => {
    const type = propertyType?.toLowerCase() || '';
    if (type.includes('apartamento') || type.includes('apartment')) {
      return "Apartamento";
    }
    if (type.includes('casa') || type.includes('house')) {
      return "Casa";
    }
    if (type.includes('comercial') || type.includes('commercial')) {
      return "Comercial";
    }
    if (type.includes('terreno') || type.includes('land')) {
      return "Terreno";
    }
    return propertyType || "Não especificado";
  };

  if (extractionStatus === 'failed') {
    return (
      <Alert variant="destructive" className="my-4">
        <AlertCircle className="h-4 w-4" />
        <AlertDescription>{statusMessage}</AlertDescription>
        {onRefresh && (
          <div className="mt-2">
            <Button variant="outline" onClick={onRefresh}>Tentar novamente</Button>
          </div>
        )}
      </Alert>
    );
  }

  console.log('[PropertyPreview] Renderizando com dados:', {
    title,
    address,
    city,
    state,
    propertyType,
    minBid,
    evaluatedValue,
    documentCount,
    bidCount
  });

  return (
    <Card className="w-full">
      <PropertyHeader 
        title={title} 
        extractionStatus={extractionStatus} 
        onRefresh={onRefresh} 
      />
      <CardContent className="space-y-6">
        {statusMessage && extractionStatus === 'partial' && (
          <Alert variant="default" className="mb-2">
            <AlertCircle className="h-4 w-4" />
            <AlertDescription>{statusMessage}</AlertDescription>
          </Alert>
        )}
        <PropertyImages images={images} title={title} />
        <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
          <div>
            <h3 className="text-lg font-semibold mb-2">Informações Básicas</h3>
            <PropertyLocation 
              address={address} 
              city={city} 
              state={state} 
              propertyType={propertyType} 
            />
            <Separator className="my-2" />
            <PropertyPricing minBid={minBid} evaluatedValue={evaluatedValue} />
          </div>
          <div>
            <h3 className="text-lg font-semibold mb-2">Informações do Leilão</h3>
            <div className="space-y-2">
              <p><strong>Tipo de Leilão:</strong> {auctionType}</p>
              <p><strong>Data do Leilão:</strong> {auctionDate || 'Data não disponível'}</p>
            </div>
            <Separator className="my-2" />
            <PropertyExtras documents={documents} auctions={auctions} />
          </div>
        </div>
        <div>
          <h3 className="text-lg font-semibold mb-2">Descrição</h3>
          <p className="text-gray-600">{description}</p>
        </div>
        <Link to={`/property/${id}`}>
          <Button size="lg" className="flex-1 w-full">
            Consultar Imóvel
          </Button>
        </Link>
      </CardContent>
    </Card>
  );
};

export default PropertyPreview; 