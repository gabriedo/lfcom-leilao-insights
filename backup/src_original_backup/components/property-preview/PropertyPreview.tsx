import React from 'react';
import { Link } from 'react-router-dom';
import { Card, CardContent } from '@/components/ui/card';
import { Button } from '@/components/ui/button';
import { Alert, AlertDescription } from '@/components/ui/alert';
import { AlertCircle } from 'lucide-react';
import { Separator } from '@/components/ui/separator';
import {
  PropertyHeader,
  PropertyImages,
  PropertyPricing,
  PropertyLocation,
  PropertyExtras
} from './index';

interface PropertyPreviewProps {
  id?: string | null;
  title: string;
  address?: string | null;
  city: string;
  state: string;
  minBid?: string | null;
  evaluatedValue: string;
  propertyType?: string | null;
  auctionType?: string | null;
  auctionDate?: string | null;
  description?: string | null;
  images?: string[] | null;
  documents?: any[] | null;
  auctions?: any[] | null;
  extractionStatus?: 'success' | 'fallback_used' | 'partial' | 'failed';
  onRefresh?: () => void;
}

const getStatusMessage = (status: string | undefined, description?: string | null) => {
  if (status === 'failed') {
    return description || 'Não foi possível extrair os dados deste imóvel. Tente novamente ou verifique a URL.';
  }
  if (status === 'partial') {
    return 'Algumas informações não foram extraídas corretamente. Confira os campos abaixo.';
  }
  return null;
};

const PropertyPreview: React.FC<PropertyPreviewProps> = (props) => {
  console.log('[PropertyPreview] props recebidas:', props);
  
  const {
    id = 'temp-id',
    title,
    address = '',
    city,
    state,
    minBid = '',
    evaluatedValue,
    propertyType = '',
    auctionType = '',
    auctionDate = '',
    description = '',
    images = [],
    documents = [],
    auctions = [],
    extractionStatus = 'success',
    onRefresh
  } = props;

  const statusMessage = getStatusMessage(extractionStatus, description);
  const documentCount = documents?.length ?? 0;
  const bidCount = auctions?.length ?? 0;

  if (extractionStatus === 'failed') {
    return (
      <Alert variant="destructive" className="my-4">
        <AlertCircle className="h-4 w-4" />
        <AlertDescription>{statusMessage}</AlertDescription>
        {onRefresh && (
          <div className="mt-2">
            <Button variant="outline" onClick={onRefresh}>Tentar novamente</Button>
          </div>
        )}
      </Alert>
    );
  }

  return (
    <Card className="w-full">
      <PropertyHeader 
        title={title || null} 
        extractionStatus={extractionStatus} 
        onRefresh={onRefresh} 
      />
      <CardContent className="space-y-6">
        {statusMessage && extractionStatus === 'partial' && (
          <Alert variant="default" className="mb-2">
            <AlertCircle className="h-4 w-4" />
            <AlertDescription>{statusMessage}</AlertDescription>
          </Alert>
        )}
        <PropertyImages 
          images={images} 
          title={title} 
        />
        <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
          <div>
            <h3 className="text-lg font-semibold mb-2">Informações Básicas</h3>
            <PropertyLocation 
              address={address} 
              city={city} 
              state={state} 
              propertyType={propertyType} 
            />
            <Separator className="my-2" />
            <PropertyPricing 
              minBid={minBid} 
              evaluatedValue={evaluatedValue} 
            />
          </div>
          <div>
            <h3 className="text-lg font-semibold mb-2">Informações do Leilão</h3>
            <div className="space-y-2">
              <p><strong>Tipo de Leilão:</strong> {auctionType || 'Não especificado'}</p>
              <p><strong>Data do Leilão:</strong> {auctionDate || 'Data não disponível'}</p>
            </div>
            <Separator className="my-2" />
            <PropertyExtras 
              documents={documents} 
              auctions={auctions} 
              documentCount={documentCount}
              bidCount={bidCount}
            />
          </div>
        </div>
        <div>
          <h3 className="text-lg font-semibold mb-2">Descrição</h3>
          <p className="text-gray-600">{description || 'Sem descrição disponível'}</p>
        </div>
        <Link to={`/property/${id}`}>
          <Button size="lg" className="flex-1 w-full">
            Consultar Imóvel
          </Button>
        </Link>
      </CardContent>
    </Card>
  );
};

export default PropertyPreview; 