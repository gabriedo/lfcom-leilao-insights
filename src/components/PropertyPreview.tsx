import React from "react";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { Badge } from "@/components/ui/badge";
import { Button } from "@/components/ui/button";
import { MapPin, Calendar, FileText, Download, ArrowUpRight, Building2, Home, Landmark } from "lucide-react";
import { format } from "date-fns";
import { ptBR } from "date-fns/locale";
import { ExtractedPropertyData } from "@/types/property";
import { Separator } from "@/components/ui/separator";

interface PropertyPreviewProps {
  data?: ExtractedPropertyData | null;
}

console.log("PropertyPreview.tsx iniciado");

export default function PropertyPreview({ data }: PropertyPreviewProps) {
  console.log("PropertyPreview data:", data);
  if (!data) {
    return null;
  }

  const {
    propertyType = '',
    address = '',
    documents = [],
    images = []
  } = data;

  const getPropertyTypeIcon = () => {
    switch (propertyType?.toLowerCase()) {
      case "apartment":
        return <Building2 className="h-4 w-4" />;
      case "house":
        return <Home className="h-4 w-4" />;
      case "land":
        return <Landmark className="h-4 w-4" />;
      default:
        return <Home className="h-4 w-4" />;
    }
  };

  const getPropertyTypeLabel = () => {
    switch (propertyType?.toLowerCase()) {
      case "apartment":
        return "Apartamento";
      case "house":
        return "Casa";
      case "land":
        return "Terreno";
      default:
        return propertyType || "Não especificado";
    }
  };

  // Função para formatar o tipo de leilão
  const formatAuctionType = (type?: string) => {
    if (!type) return 'Não informado';
    const types: { [key: string]: string } = {
      'judicial': 'Leilão Judicial',
      'extrajudicial': 'Leilão Extrajudicial',
      'bank': 'Venda Direta (Banco)',
      'other': 'Outro'
    };
    return types[type.toLowerCase()] || type;
  };

  return (
    <div className="space-y-6">
      {/* Imagens do imóvel */}
      {images && images.length > 0 && (
        <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
          {images.map((image, index) => (
            <div key={index} className="relative aspect-video rounded-lg overflow-hidden">
              <img
                src={image}
                alt={`Imagem ${index + 1}`}
                className="w-full h-full object-cover"
                loading="lazy"
              />
            </div>
          ))}
        </div>
      )}

      {/* Informações principais */}
      <Card>
        <CardHeader>
          <CardTitle className="flex items-center gap-2">
            {getPropertyTypeIcon()}
            {getPropertyTypeLabel()}
          </CardTitle>
          {address && <p className="text-sm text-muted-foreground">{address}</p>}
        </CardHeader>
        <CardContent className="space-y-4">
          {documents?.length > 0 && (
            <>
              <Separator />
              <div className="space-y-2">
                <p className="text-sm font-medium">Documentos</p>
                <div className="grid gap-2">
                  {documents.map((doc, index) => (
                    doc?.url && (
                      <div key={index} className="flex items-center gap-2">
                        <FileText className="h-4 w-4 text-muted-foreground" />
                        <a
                          href={doc.url}
                          target="_blank"
                          rel="noopener noreferrer"
                          className="text-sm text-blue-600 hover:underline"
                        >
                          {doc.name || `Documento ${index + 1}`}
                        </a>
                        <Badge variant="outline" className="ml-auto">
                          {doc.type?.toUpperCase() || "PDF"}
                        </Badge>
                      </div>
                    )
                  ))}
                </div>
              </div>
            </>
          )}

          {/* Valores */}
          <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
            <div className="space-y-2">
              <p className="text-sm text-muted-foreground">Lance Mínimo</p>
              <p className="text-2xl font-bold text-primary">
                {data.minBid || 'Não informado'}
              </p>
            </div>
            <div className="space-y-2">
              <p className="text-sm text-muted-foreground">Valor de Avaliação</p>
              <p className="text-2xl font-bold">
                {data.evaluatedValue || 'Não informado'}
              </p>
            </div>
          </div>

          {/* Data do leilão */}
          {data.auctionDate && (
            <div className="space-y-2">
              <div className="flex items-center text-muted-foreground">
                <Calendar className="h-4 w-4 mr-2" />
                <span className="text-sm">Data do Leilão</span>
              </div>
              <p className="text-lg">
                {format(new Date(data.auctionDate), "PPP", { locale: ptBR })}
              </p>
            </div>
          )}

          {/* Descrição */}
          {data.description && (
            <div className="space-y-2">
              <p className="text-sm text-muted-foreground">Descrição</p>
              <p className="text-lg">{data.description}</p>
            </div>
          )}
        </CardContent>
      </Card>
    </div>
  );
} 