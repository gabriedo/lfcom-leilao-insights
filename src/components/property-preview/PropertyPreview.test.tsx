import React from 'react';
import { describe, it, expect, vi, afterEach } from 'vitest';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import PropertyPreview from '../PropertyPreview';
import { MemoryRouter } from 'react-router-dom';

// Limpa o DOM após cada teste
afterEach(cleanup);

const baseProps = {
  id: '1',
  title: 'Apartamento Teste',
  address: 'Rua Exemplo, 123',
  city: 'São Paulo',
  state: 'SP',
  minBid: 'R$ 100.000,00',
  evaluatedValue: 'R$ 120.000,00',
  propertyType: 'Apartamento',
  auctionType: 'Judicial',
  auctionDate: '2025-05-10',
  description: 'Descrição do imóvel',
  images: ['https://img.com/1.jpg'],
  documents: [{ name: 'Edital' }],
  auctions: [{ label: '1º Leilão' }],
  extractionStatus: 'success' as 'success',
  onRefresh: vi.fn(),
};

function renderComponent(customProps = {}) {
  return render(
    <MemoryRouter>
      <PropertyPreview {...baseProps} {...customProps} />
    </MemoryRouter>
  );
}

describe('PropertyPreview (integração)', () => {
  it('renderiza corretamente com status "success"', () => {
    renderComponent();
    expect(screen.getByText('Apartamento Teste')).toBeInTheDocument();
    expect(screen.getByText('Rua Exemplo, 123')).toBeInTheDocument();
    expect(screen.getByText((content) => content.includes('São Paulo') && content.includes('SP'))).toBeInTheDocument();
    expect(screen.getByText('Apartamento')).toBeInTheDocument();
    expect(screen.getByText('R$ 100.000,00')).toBeInTheDocument();
    expect(screen.getByText('R$ 120.000,00')).toBeInTheDocument();
    expect(screen.getByText('Judicial')).toBeInTheDocument();
    expect(screen.getByText('2025-05-10')).toBeInTheDocument();
    expect(screen.getByText('Descrição do imóvel')).toBeInTheDocument();
    expect(screen.getByRole('img')).toHaveAttribute('src', 'https://img.com/1.jpg');
    expect(screen.getByRole('button', { name: /consultar imóvel/i })).toBeInTheDocument();
    expect(screen.getByText('Sucesso')).toBeInTheDocument();
  });

  it('renderiza fallback parcial com status "partial"', () => {
    renderComponent({
      extractionStatus: 'partial' as 'partial',
      title: '',
      address: '',
      city: '',
      state: '',
      minBid: '',
      evaluatedValue: '',
      propertyType: '',
      auctionType: '',
      auctionDate: '',
      description: '',
      images: [],
      documents: [],
      auctions: [],
    });
    expect(screen.getByText(/algumas informações não foram extraídas/i)).toBeInTheDocument();
    const notSpecifieds = screen.getAllByText('Não especificado');
    expect(notSpecifieds.length).toBeGreaterThanOrEqual(2);
    expect(screen.getByText('Parcial')).toBeInTheDocument();
    expect(screen.getByText('Sem descrição disponível')).toBeInTheDocument();
    expect(screen.getByText('Data não disponível')).toBeInTheDocument();
  });

  it('renderiza alerta de falha com status "failed"', () => {
    const onRefresh = vi.fn();
    renderComponent({ extractionStatus: 'failed' as 'failed', description: 'Erro ao extrair', onRefresh });
    expect(screen.getByText(/erro ao extrair/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /tentar novamente/i })).toBeInTheDocument();
    fireEvent.click(screen.getByRole('button', { name: /tentar novamente/i }));
    expect(onRefresh).toHaveBeenCalledTimes(1);
  });

  it('renderiza corretamente com status "fallback_used"', () => {
    renderComponent({ extractionStatus: 'fallback_used' as 'fallback_used' });
    expect(screen.getByText('Fallback')).toBeInTheDocument();
    expect(screen.getByText('Apartamento Teste')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /consultar imóvel/i })).toBeInTheDocument();
  });

  it('renderiza corretamente com props ausentes ou nulas', () => {
    renderComponent({
      id: null,
      title: null,
      address: null,
      city: null,
      state: null,
      minBid: null,
      evaluatedValue: null,
      propertyType: null,
      auctionType: null,
      auctionDate: null,
      description: null,
      images: undefined,
      documents: undefined,
      auctions: undefined,
      extractionStatus: 'success' as 'success',
      onRefresh: undefined,
    });
    const notSpecifieds = screen.getAllByText('Não especificado');
    expect(notSpecifieds.length).toBeGreaterThanOrEqual(1);
    expect(screen.getByText('Sem descrição disponível')).toBeInTheDocument();
    expect(screen.getByText('Data não disponível')).toBeInTheDocument();
    expect(screen.getByText('Sucesso')).toBeInTheDocument();
  });

  it('chama onRefresh ao clicar no botão de atualizar', () => {
    const onRefresh = vi.fn();
    renderComponent({ extractionStatus: 'partial' as 'partial', onRefresh });
    const btn = screen.getByRole('button', { name: /atualizar|refresh|tentar novamente/i });
    fireEvent.click(btn);
    expect(onRefresh).toHaveBeenCalledTimes(1);
  });
}); 