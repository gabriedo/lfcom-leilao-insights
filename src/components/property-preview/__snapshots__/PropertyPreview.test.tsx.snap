// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PropertyPreview (integração) > renderiza corretamente apenas com title, minBid e images 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm w-full"
  >
    <div
      class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
    >
      <h3
        class="tracking-tight text-2xl font-bold truncate max-w-[70vw]"
      >
        Mega Leilão - Imóvel Teste
      </h3>
      <div
        class="flex items-center space-x-2"
      >
        <div
          class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground hover:bg-primary/80"
        >
          Sucesso
        </div>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10"
          title="Atualizar dados"
        >
          <svg
            class="lucide lucide-refresh-cw h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 12a9 9 0 0 1 9-9 9.75 9.75 0 0 1 6.74 2.74L21 8"
            />
            <path
              d="M21 3v5h-5"
            />
            <path
              d="M21 12a9 9 0 0 1-9 9 9.75 9.75 0 0 1-6.74-2.74L3 16"
            />
            <path
              d="M8 16H3v5"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="p-6 pt-0 space-y-6"
    >
      <div
        class="relative aspect-video rounded-lg overflow-hidden"
      >
        <img
          alt="Mega Leilão - Imóvel Teste"
          class="w-full h-full object-cover"
          loading="lazy"
          src="https://img.com/mega.jpg"
        />
      </div>
      <div
        class="grid grid-cols-1 md:grid-cols-2 gap-4"
      >
        <div>
          <h3
            class="text-lg font-semibold mb-2"
          >
            Informações Básicas
          </h3>
          <div
            class="space-y-2"
          >
            <p>
              <strong>
                Endereço:
              </strong>
               Endereço não disponível
            </p>
            <p>
              <strong>
                Cidade:
              </strong>
               Cidade não disponível - Estado não disponível
            </p>
            <p>
              <strong>
                Tipo:
              </strong>
               
              <svg
                class="lucide lucide-home h-4 w-4 inline-block mr-1"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m3 9 9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
                />
                <polyline
                  points="9 22 9 12 15 12 15 22"
                />
              </svg>
              Não especificado
            </p>
          </div>
          <div
            class="shrink-0 bg-border h-[1px] w-full my-2"
            data-orientation="horizontal"
            role="none"
          />
          <div
            class="space-y-2"
          >
            <p>
              <strong>
                Lance Mínimo:
              </strong>
               R$ 200.000,00
            </p>
            <p>
              <strong>
                Valor Avaliado:
              </strong>
               Não disponível
            </p>
          </div>
        </div>
        <div>
          <h3
            class="text-lg font-semibold mb-2"
          >
            Informações do Leilão
          </h3>
          <div
            class="space-y-2"
          >
            <p>
              <strong>
                Tipo de Leilão:
              </strong>
               Não especificado
            </p>
            <p>
              <strong>
                Data do Leilão:
              </strong>
               Data não disponível
            </p>
          </div>
          <div
            class="shrink-0 bg-border h-[1px] w-full my-2"
            data-orientation="horizontal"
            role="none"
          />
          <div
            class="space-y-2"
          >
            <div>
              <strong>
                Documentos:
              </strong>
               0 disponível(is)
            </div>
            <div
              class="shrink-0 bg-border h-[1px] w-full"
              data-orientation="horizontal"
              role="none"
            />
            <div>
              <strong>
                Lances:
              </strong>
               0 registrado(s)
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="text-lg font-semibold mb-2"
        >
          Descrição
        </h3>
        <p
          class="text-gray-600"
        >
          Sem descrição disponível
        </p>
      </div>
      <a
        href="/property/mega-1"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-11 rounded-md px-8 flex-1 w-full"
        >
          Consultar Imóvel
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PropertyPreview (integração) > renderiza corretamente com snapshot 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm w-full"
  >
    <div
      class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
    >
      <h3
        class="tracking-tight text-2xl font-bold truncate max-w-[70vw]"
      >
        Apartamento Teste
      </h3>
      <div
        class="flex items-center space-x-2"
      >
        <div
          class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground hover:bg-primary/80"
        >
          Sucesso
        </div>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10"
          title="Atualizar dados"
        >
          <svg
            class="lucide lucide-refresh-cw h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 12a9 9 0 0 1 9-9 9.75 9.75 0 0 1 6.74 2.74L21 8"
            />
            <path
              d="M21 3v5h-5"
            />
            <path
              d="M21 12a9 9 0 0 1-9 9 9.75 9.75 0 0 1-6.74-2.74L3 16"
            />
            <path
              d="M8 16H3v5"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="p-6 pt-0 space-y-6"
    >
      <div
        class="relative aspect-video rounded-lg overflow-hidden"
      >
        <img
          alt="Apartamento Teste"
          class="w-full h-full object-cover"
          loading="lazy"
          src="https://img.com/1.jpg"
        />
      </div>
      <div
        class="grid grid-cols-1 md:grid-cols-2 gap-4"
      >
        <div>
          <h3
            class="text-lg font-semibold mb-2"
          >
            Informações Básicas
          </h3>
          <div
            class="space-y-2"
          >
            <p>
              <strong>
                Endereço:
              </strong>
               Rua Exemplo, 123
            </p>
            <p>
              <strong>
                Cidade:
              </strong>
               São Paulo - SP
            </p>
            <p>
              <strong>
                Tipo:
              </strong>
               
              <svg
                class="lucide lucide-building2 h-4 w-4 inline-block mr-1"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6 22V4a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v18Z"
                />
                <path
                  d="M6 12H4a2 2 0 0 0-2 2v6a2 2 0 0 0 2 2h2"
                />
                <path
                  d="M18 9h2a2 2 0 0 1 2 2v9a2 2 0 0 1-2 2h-2"
                />
                <path
                  d="M10 6h4"
                />
                <path
                  d="M10 10h4"
                />
                <path
                  d="M10 14h4"
                />
                <path
                  d="M10 18h4"
                />
              </svg>
              Apartamento
            </p>
          </div>
          <div
            class="shrink-0 bg-border h-[1px] w-full my-2"
            data-orientation="horizontal"
            role="none"
          />
          <div
            class="space-y-2"
          >
            <p>
              <strong>
                Lance Mínimo:
              </strong>
               R$ 100.000,00
            </p>
            <p>
              <strong>
                Valor Avaliado:
              </strong>
               R$ 120.000,00
            </p>
          </div>
        </div>
        <div>
          <h3
            class="text-lg font-semibold mb-2"
          >
            Informações do Leilão
          </h3>
          <div
            class="space-y-2"
          >
            <p>
              <strong>
                Tipo de Leilão:
              </strong>
               Judicial
            </p>
            <p>
              <strong>
                Data do Leilão:
              </strong>
               2025-05-10
            </p>
          </div>
          <div
            class="shrink-0 bg-border h-[1px] w-full my-2"
            data-orientation="horizontal"
            role="none"
          />
          <div
            class="space-y-2"
          >
            <div>
              <strong>
                Documentos:
              </strong>
               1 disponível(is)
              <div
                class="flex flex-wrap gap-2 mt-1"
              >
                <div
                  class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground"
                >
                  Edital
                </div>
              </div>
            </div>
            <div
              class="shrink-0 bg-border h-[1px] w-full"
              data-orientation="horizontal"
              role="none"
            />
            <div>
              <strong>
                Lances:
              </strong>
               1 registrado(s)
              <div
                class="flex flex-wrap gap-2 mt-1"
              >
                <div
                  class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80"
                >
                  1º Leilão
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <h3
          class="text-lg font-semibold mb-2"
        >
          Descrição
        </h3>
        <p
          class="text-gray-600"
        >
          Descrição do imóvel
        </p>
      </div>
      <a
        href="/property/1"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-11 rounded-md px-8 flex-1 w-full"
        >
          Consultar Imóvel
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
